tests/unit/components/frost-checkbox-test.js

```diff
@@ -17,53 +27,69 @@ describeComponent(
       component = this.subject()
     })
 
-    it('sets dependent keys correctly', function () {
-      const isCheckedDependentKeys = [
-        'checked'
-      ]
+    it('includes className frost-button', function () {
+      expect(component.classNames).to.include('frost-checkbox')
+    })
 
-      const sizeClassDependentKeys = [
-        'size'
-      ]
+    it('sets default property values correctly', function () {
+      expect(
+        component.get('size'),
+        'size: "small"'
+      ).to.eql('small')
 
-      const inputIdDependentKeys = [
-        'id'
-      ]
+      expect(
+        component.get('label'),
+        'label: ""'
+      ).to.eql('')
+
+      expect(
+        component.get('autofocus'),
+        'autofocus: "false"'
+      ).to.be.false
 
       expect(
-        component.isChecked._dependentKeys,
-        'Dependent keys are correct for isChecked()'
-      ).to.eql(isCheckedDependentKeys)
+        component.get('checked'),
+        'checked: "false"'
+      ).to.be.false
 
       expect(
-        component.sizeClass._dependentKeys,
-        'Dependent keys are correct for sizeClass()'
-      ).to.eql(sizeClassDependentKeys)
+        component.get('disabled'),
+        'disabled: "false"'
+      ).to.be.false
+
+      expect(
+        component.get('hook'),
+        'hook: "undefined"'
+      ).to.be.undefined
+    })
+
+    it('sets dependent keys correctly', function () {
+      const inputIdDependentKeys = [
+        'id'
+      ]
 
       expect(
         component.inputId._dependentKeys,
         'Dependent keys are correct for inputId()'
       ).to.eql(inputIdDependentKeys)
     })
 
-    it('defaults state to unchecked', function () {
-      expect(component.get('isChecked')).to.equal(false)
+    it('has the expected Mixins', function () {
+      expect(
+        PropTypeMixin.detect(component),
+        'PropTypeMixin Mixin is present'
+      ).to.be.true
     })
 
-    describe('isChecked', function () {
-      [
-        {in: undefined, out: false},
-        {in: null, out: false},
-        {in: false, out: false},
-        {in: true, out: true}
-      ].forEach((test) => {
-        it(`returns ${test.out} when checked is ${test.in}`, function () {
-          run(() => {
-            component.set('checked', test.in)
-          })
-          expect(component.get('isChecked')).to.equal(test.out)
-        })
-      })
+    it('sets the inputId', function () {
+      const testId = 99
+
+      run(() => component.set('id', testId))
```

Comment:
With the presence of hooks, I'm not sure we should even really bother with ids anymore - just treat them as controlled by Ember and keep our hands off